cartographer/mapping/pose_extrapolator.h

```diff
@@ -49,6 +49,16 @@ class PoseExtrapolator {
   void AddOdometryData(const sensor::OdometryData& odometry_data);
   transform::Rigid3d ExtrapolatePose(common::Time time);
 
+  // Must be called after the first IMU data and no pose were added. Estimates
+  // an initial gravity aligned orientation from the linear acceleration of the
+  // first IMU data.
+  void AddInitialGravityAlignedPose();
```

Comment:
I addressed the second issue. I'll change 2D in a follow up PR since this will slightly change 2D SLAM behavior.